ports/ffmpeg/0006-fix-StaticFeatures.patch

```
@@ -2,19 +2,48 @@ diff --git a/configure b/configure
+ enabled libmodplug        && require_pkg_config libmodplug libmodplug libmodplug/modplug.h ModPlug_Load
+-enabled libmp3lame        && require "libmp3lame >= 3.98.3" lame/lame.h lame_set_VBR_quality -lmp3lame $libm_extralibs
++enabled libmp3lame        && { check_lib libmp3lame lame/lame.h lame_set_VBR_quality -lmp3lame $libm_extralibs || 
++                               require libmp3lame lame/lame.h lame_set_VBR_quality -llibmp3lame-static -llibmpghip-static $libm_extralibs; }
+ enabled libmysofa         && { check_pkg_config libmysofa libmysofa mysofa.h mysofa_load ||
+                                require libmysofa mysofa.h mysofa_load -lmysofa $zlib_extralibs; }
+ enabled libnpp            && { check_lib libnpp npp.h nppGetLibVersion -lnppig -lnppicc -lnppc -lnppidei ||
+@@ -6292,7 +6293,7 @@ if enabled debug_configure; then
+ else
+     enabled libsnappy     && require libsnappy snappy-c.h snappy_compress -lsnappy -lstdc++
+ fi
+-enabled libsoxr           && require libsoxr soxr.h soxr_create -lsoxr
++enabled libsoxr           && require libsoxr soxr.h soxr_create -lsoxr -lm
+ enabled libssh            && require_pkg_config libssh libssh libssh/sftp.h sftp_init
+ enabled libspeex          && require_pkg_config libspeex speex speex/speex.h speex_decoder_init
+ enabled libsrt            && require_pkg_config libsrt "srt >= 1.3.0" srt/srt.h srt_socket
+@@ -6360,6 +6361,8 @@ enabled openal            && { { for al_extralibs in "${OPENAL_LIBS}" "-lopenal"
  enabled opencl            && { check_pkg_config opencl OpenCL CL/cl.h clEnqueueNDRangeKernel ||
                                 check_lib opencl OpenCL/cl.h clEnqueueNDRangeKernel -Wl,-framework,OpenCL ||
                                 check_lib opencl CL/cl.h clEnqueueNDRangeKernel -lOpenCL ||
 +                               check_lib opencl CL/cl.h clEnqueueNDRangeKernel -lOpenCL -lAdvapi32 -lOle32 -lCfgmgr32||
```

Advapi32 only support windows.

---

Yes, so in the non-window case this check_lib call will fail as those libs dont exist. This is intended as the check_lib call is used to detect if the lib is available. So on windows it will succeed and then add the required libs to the final build. However, on non-windows it will silently fail and do nothing. This is intended behavior as one of the existing other check_lib calls will succeed on linux and add the required libs for that case.